presto-native-execution/presto_cpp/main/tests/ServerOperationTest.cpp

```diff
@@ -0,0 +1,92 @@
+ * limitations under the License.
+ */
+#include "presto_cpp/main/ServerOperation.h"
+#include <gtest/gtest.h>
+#include "velox/common/base/Exceptions.h"
+
+namespace facebook::presto {
+
+class ServerOperationTest : public testing::Test {};
+
+TEST_F(ServerOperationTest, targetActionLookup) {
+  {
+    // Targets lookup verification
+    EXPECT_EQ(
+        ServerOperation::kTargetLookup.size(),
+        ServerOperation::kReverseTargetLookup.size());
+    const auto endReverseIt = ServerOperation::kReverseTargetLookup.end();
+    for (auto lookupIt = ServerOperation::kTargetLookup.begin();
+         lookupIt != ServerOperation::kTargetLookup.end();
+         lookupIt++) {
+      EXPECT_NE(
+          ServerOperation::kReverseTargetLookup.find(lookupIt->second),
+          endReverseIt);
+    }
+    const auto endLookupIt = ServerOperation::kTargetLookup.end();
+    for (auto reverseIt = ServerOperation::kReverseTargetLookup.begin();
+         reverseIt != ServerOperation::kReverseTargetLookup.end();
+         reverseIt++) {
+      EXPECT_NE(
+          ServerOperation::kTargetLookup.find(reverseIt->second), endLookupIt);
+    }
+  }
+
+  {
+    // Actions lookup verification
+    EXPECT_EQ(
+        ServerOperation::kActionLookup.size(),
+        ServerOperation::kReverseActionLookup.size());
+    const auto endReverseIt = ServerOperation::kReverseActionLookup.end();
+    for (auto lookupIt = ServerOperation::kActionLookup.begin();
+         lookupIt != ServerOperation::kActionLookup.end();
+         lookupIt++) {
+      EXPECT_NE(
+          ServerOperation::kReverseActionLookup.find(lookupIt->second),
+          endReverseIt);
+    }
+    const auto endLookupIt = ServerOperation::kActionLookup.end();
+    for (auto reverseIt = ServerOperation::kReverseActionLookup.begin();
+         reverseIt != ServerOperation::kReverseActionLookup.end();
+         reverseIt++) {
+      EXPECT_NE(
+          ServerOperation::kActionLookup.find(reverseIt->second), endLookupIt);
+    }
+  }
+}
+
+TEST_F(ServerOperationTest, stringEnumConversion) {
+  for (auto lookupIt = ServerOperation::kTargetLookup.begin();
+       lookupIt != ServerOperation::kTargetLookup.end();
+       lookupIt++) {
+    EXPECT_EQ(
+        ServerOperation::targetFromString(lookupIt->first), lookupIt->second);
+    EXPECT_EQ(ServerOperation::targetString(lookupIt->second), lookupIt->first);
+  }
+  EXPECT_THROW(
+      ServerOperation::targetFromString("UNKNOWN_TARGET"),
+      velox::VeloxUserError);
+
+  for (auto lookupIt = ServerOperation::kActionLookup.begin();
+       lookupIt != ServerOperation::kActionLookup.end();
+       lookupIt++) {
+    EXPECT_EQ(
+        ServerOperation::actionFromString(lookupIt->first), lookupIt->second);
+    EXPECT_EQ(ServerOperation::actionString(lookupIt->second), lookupIt->first);
+  }
+  EXPECT_THROW(
+      ServerOperation::actionFromString("UNKNOWN_ACTION"),
+      velox::VeloxUserError);
+}
+
```

Comment:
I think we might consider to add PrestoServerTest.cpp to allow us to setup a presto cpp server and run test on top of that as a coordinator in followup.

Reply:
We sort of have it in Java integration tests